common/net/conn/conn.go

```diff
@@ -0,0 +1,121 @@
+package conn
+
+import (
+	"io"
+	"net"
+)
+
+// MaxBufferSize - size which can store about 1024 frames of AES256
+const MaxBufferSize = 32768
+
+// MinBufferFrameSet - can contains 32 frames of AES256
+const MinBufferFrameSet = 1024
+
+// Conn represents a secured connection.
+// which implements the net.Conn interface.
+type Conn struct {
+	conn   net.Conn
+	crypto Crypto
+	buffer []byte
+}
+
+// NewClientConn - Creates Client that should use to connect to a server
+func NewClientConn(conn net.Conn) *Conn {
```

Comment:
There must be something like
https://github.com/pastelnetwork/gonode/blob/21f2cbb1bb2c4dacf9440099a16fafbb180076fd/common/net/conn.go#L53